pandas/_libs/tslibs/strptime.pyx

```diff
@@ -589,8 +588,8 @@ class TimeRE(dict):
         else:
             return ''
         regex = '|'.join(re.escape(stuff) for stuff in to_convert)
-        regex = '(?P<%s>%s' % (directive, regex)
-        return '%s)' % regex
+        regex = f'(?P<{directive}>{regex}'
+        return f'{regex})'
```

Comment:
can you add the closing paren on the previous line

Reply:
And returning ```regex``` not ```str(regex)``` right?